photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java

```diff
@@ -0,0 +1,139 @@
+    private Map<Integer, Pose3d> aprilTags;
+    private PoseStrategy strategy;
+    private ArrayList<Pair<PhotonCamera, Transform3d>> cameras;
+    private Pose3d lastPose;
+    
+    /**
+     * Create a new RobotPoseEstimator. 
+     * <p>
+     * Example: 
+     *  <code>
+     *  <p>
+     *  Map<Integer, Pose3d> map = new HashMap<>();
+     *  <p>
+     *  map.put(1, new Pose3d(1.0, 2.0, 3.0, new Rotation3d())); // Tag ID 1 is at (1.0,2.0,3.0)
+     *  </code>
+     * 
+     * @param aprilTags A Map linking AprilTag IDs to Pose3ds with respect to the FIRST field.
+     * @param strategy The strategy it should use to determine the best pose.
+     * @param cameras An ArrayList of Pairs of PhotonCameras and their respective Transform3ds from the center of the robot to the cameras. 
+     */
+    public RobotPoseEstimator(Map<Integer, Pose3d> aprilTags, PoseStrategy strategy, ArrayList<Pair<PhotonCamera, Transform3d>> cameras) {
+        this.aprilTags = aprilTags;
+        this.strategy = strategy;
+        this.cameras = cameras;
+    }
+
+
+    public Pose3d update() {
+        if (cameras.isEmpty()) {
+            DriverStation.reportError("[RobotPoseEstimator] Missing any camera!", false);
+            return lastPose;
+        }
+        switch (strategy) {
+            case LOWEST_AMBIGUITY:
+                lastPose = lowestAmbiguityStrategy();
+                return lastPose;
+            case CLOSEST_TO_CAMERA_HEIGHT:
+
+                return lastPose;
+            default:
+                DriverStation.reportError("[RobotPoseEstimator] Invalid pose strategy!", false);
+                return lastPose;
+        }
+    }
+
+    private Pose3d lowestAmbiguityStrategy() {
+        // Loop over each ambiguity of all the cameras
+        int lowestAI = -1;
+        int lowestAJ = -1;
+        double lowestAmbiguityScore = 10;
+        for (int i = 0; i < cameras.size(); i++) {
+            Pair<PhotonCamera, Transform3d> p = cameras.get(i);
+            List<PhotonTrackedTarget> targets = p.getFirst().getLatestResult().targets;
+            for (int j = 0; j < targets.size(); j++) {
+                if (targets.get(j).getPoseAmbiguity() > lowestAmbiguityScore) {
+                    lowestAI = i;
+                    lowestAJ = j;
+                    lowestAmbiguityScore = targets.get(j).getPoseAmbiguity();
+                }
+            }
+        }
+
+        // No targets, return the last pose
+        if (lowestAI == -1 || lowestAJ == -1) {
+            return lastPose;
+        }
+
+        // Pick the lowest and do the heavy calculations
+        PhotonTrackedTarget bestTarget = cameras.get(lowestAI).getFirst().getLatestResult().targets.get(lowestAJ);
+
+        // If the map doesn't contain the ID fail
+        if (!aprilTags.containsKey(bestTarget.getFiducialId())) {
+            DriverStation.reportError(
+                    "[RobotPoseEstimator] Tried to get pose of unknown April Tag: " + bestTarget.getFiducialId(),
```

Comment:
Possibly...
The reason for putting it as I foresaw teams trying to debug why it wasn't working when the issue is they are using a tag that isn't defined.

Thoughts on changing it so it will only print once per tag? Can be accomplished with a hashset.

Reply:
Once per tag sounds good tome.